pkg/build/strategies/sti/sti_test.go

```diff
@@ -549,60 +552,34 @@ func TestSaveArtifactsExtractError(t *testing.T) {
 
 func TestFetchSource(t *testing.T) {
 	type fetchTest struct {
-		validCloneSpec   bool
 		refSpecified     bool
 		cloneExpected    bool
 		checkoutExpected bool
-		copyExpected     bool
 		sourcePath       string
 		expectedError    *error
 	}
 
-	err := stierr.NewSourcePathError("error")
 	tests := []fetchTest{
 		// 0
 		{
-			validCloneSpec:   false,
-			refSpecified:     false,
-			cloneExpected:    false,
-			checkoutExpected: false,
-			copyExpected:     false,
-			sourcePath:       "invalid/path",
-			expectedError:    &err,
-		},
-		// 1
-		{
-			validCloneSpec:   false,
-			refSpecified:     false,
-			cloneExpected:    false,
-			checkoutExpected: false,
-			copyExpected:     true,
```

Comment:
These tests were exercising a code path in pkg/scm/git.(*Clone).Download which I believe was no longer reachable in practice (only by this test) and was somewhat rotten.  I removed the code path and the test.  My logic is: a git.Clone object is only created in scm.go after ValidCloneSpec has returned true.  git.(*Clone).Download had a code path to deal with what happens if ValidCloneSpec later returns false.  I have retained an instrumented call to ValidCloneSpec in git.(*Clone).Download to log an error if this happens, but have otherwise deleted the erronenous code path.